libraries/AP_ExternalAHRS/AP_ExternalAHRS_VectorNav.cpp

```diff
@@ -551,7 +551,28 @@ void AP_ExternalAHRS_VectorNav::process_packet2(const uint8_t *b)
     // get ToW in milliseconds
     gps.gps_week = pkt2.timeGPS / (AP_MSEC_PER_WEEK * 1000000ULL);
     gps.ms_tow = (pkt2.timeGPS / 1000000ULL) % (60*60*24*7*1000ULL);
-    gps.fix_type = pkt2.GPS1Fix;
+    switch (pkt2.GPS1Fix) {
+    case 0:
+    case 1:
+        gps.fix_type = AP_GPS::NO_FIX;
+        break;
+    case 2:
+        gps.fix_type = AP_GPS::GPS_OK_FIX_2D;
+        break;
+    case 3:
+        gps.fix_type = AP_GPS::GPS_OK_FIX_3D;
+        break;
+    case 4:
+    case 5:
+        gps.fix_type = AP_GPS::GPS_OK_FIX_3D_DGPS;
+        break;
+    case 6:
```

Comment:
Looks like case 6 and case 7 should be moved to case 7 and case 8.

RTK float is 7, RTK fix is 8 as reported by VN's GPS Fix output

Reply:
5 means diff which is sometimes output (mostly when SBAS isn't available but they're passing some RTCM). 6 is PPP which can technically be output, but only if they're passing us an external receiver which is equipped with PPP, and we're echoing that through to ardupilot.